bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeAirDevice.java

```diff
@@ -0,0 +1,530 @@
+    }
+
+    public void setDeviceDry(DatagramSocket clientSocket, int value) throws GreeException {
+        setCommandValue(clientSocket, GREE_PROP_DRY, value);
+    }
+
+    public void setDeviceHealth(DatagramSocket clientSocket, int value) throws GreeException {
+        setCommandValue(clientSocket, GREE_PROP_HEALTH, value);
+    }
+
+    public void setDevicePwrSaving(DatagramSocket clientSocket, int value) throws GreeException {
+        // Set the values in the HashMap
+        HashMap<String, Integer> parameters = new HashMap<>();
+        parameters.put(GREE_PROP_PWR_SAVING, value);
+        parameters.put(GREE_PROP_WINDSPEED, 0);
+        parameters.put(GREE_PROP_QUIET, 0);
+        parameters.put(GREE_PROP_TURBO, 0);
+        parameters.put(GREE_PROP_SLEEP, 0);
+        parameters.put(GREE_PROP_SLEEPMODE, 0);
+        executeCommand(clientSocket, parameters);
+    }
+
+    public int getIntStatusVal(String valueName) {
+        /*
+         * Note : Values can be:
+         * "Pow": Power (0 or 1)
+         * "Mod": Mode: Auto: 0, Cool: 1, Dry: 2, Fan: 3, Heat: 4
+         * "SetTem": Requested Temperature
+         * "WdSpd": Fan Speed : Low:1, Medium Low:2, Medium :3, Medium High :4, High :5
+         * "Air": Air Mode Enabled
+         * "Blo": Dry
+         * "Health": Health
+         * "SwhSlp": Sleep
+         * "SlpMod": ???
+         * "Lig": Light On
+         * "SwingLfRig": Swing Left Right
+         * "SwUpDn": Swing Up Down: // Ceiling:0, Upwards : 10, Downwards : 11, Full range : 1
+         * "Quiet": Quiet mode
+         * "Tur": Turbo
+         * "StHt": 0,
+         * "TemUn": Temperature unit, 0 for Celsius, 1 for Fahrenheit
+         * "HeatCoolType"
+         * "TemRec": (0 or 1), Send with SetTem, when TemUn==1, distinguishes between upper and lower integer Fahrenheit
+         * temp
+         * "SvSt": Power Saving
+         */
+        // Find the valueName in the Returned Status object
+        if (isStatusAvailable()) {
+            List<String> colList = Arrays.asList(statusResponseGson.get().packJson.cols);
+            List<Integer> valList = Arrays.asList(statusResponseGson.get().packJson.dat);
+            int valueArrayposition = colList.indexOf(valueName);
+            if (valueArrayposition != -1) {
+                // get the Corresponding value
+                Integer value = valList.get(valueArrayposition);
+                return value;
+            }
+        }
+
+        return -1;
+    }
+
+    public boolean isStatusAvailable() {
+        return statusResponseGson.isPresent();
+    }
+
+    public boolean hasStatusValChanged(String valueName) throws GreeException {
+        if (!prevStatusResponsePackGson.isPresent()) {
+            return true; // update value if there is no previous one
+        }
+        // Find the valueName in the Current Status object
+        List<String> currcolList = Arrays.asList(statusResponseGson.get().packJson.cols);
```

Comment:
if statusResponseGson isn't present then this will throw an exception.

Reply:
I suppose as it is designed that wouldn't happen so feel free to ignore this change.